### PR TITLE
log-backup: calculate global checkpoint-ts from storage

### PR DESCRIPTION
**Pull request by joccau**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/36246

Problem Summary:

### What is changed and how it works?
- calcuate global checkpoint-ts from the files written by `tikv` periodically.
- This PR is associated with https://github.com/tikv/tikv/pull/13035
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

### manual test
- ./bin/br log status -u "X.X.X.X:2379"
● Total 1 Tasks.
> #1 <
              name: store-by-storeid
            status: ● NORMAL
             start: 2022-07-14 20:08:03.268 +0800 CST
               end: 2090-11-18 22:07:45.624 +0800 CST
           storage: s3://tmp/store-by-storeid/log1
       speed(est.): 0.00 ops/s
checkpoint[global]: 2022-07-17 19:36:27.418 +0800 CST; gap=1m11s
- ./bin/br log metadata -s "XXXX"
[2022/07/17 19:37:38.659 +08:00] [INFO] [collector.go:69] ["log metadata"] [log-min-ts=434582449885806593] [log-min-date="2022-07-14 20:08:03.268 +0800 CST"] [log-max-ts=434649900624904193] [log-max-date="2022-07-17 19:36:27.418 +0800 CST"]

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/task/stream.go [hunk 1]
```diff
@@ -1273,6 +1274,22 @@ func getLogRange(
 	return logMinTS, logMaxTS, nil
 }
 
+func getGlobalCheckpointFromStorage(ctx context.Context, s storage.ExternalStorage) (uint64, error) {
+	var globalCheckPointTS uint64 = 0
+	opt := storage.WalkOption{SubDir: stream.GetStreamBackupGlobalCheckpointPrefix()}
+	err := s.WalkDir(ctx, &opt, func(path string, size int64) error {
+		buff, err := s.ReadFile(ctx, path)
```
Comment (3pointer) [inline br/pkg/task/stream.go:1281]: maybe check `.ts` suffix.

**Review by Leavrth — APPROVED** [review]

File: br/pkg/task/stream.go [hunk 2]
```diff
@@ -1264,7 +1265,7 @@ func getLogRange(
 	logMinTS := mathutil.Max(logStartTS, truncateTS)
 
 	// get max global resolved ts from metas.
-	logMaxTS, err := getGlobalResolvedTS(ctx, s)
```
Comment (Leavrth) [inline br/pkg/task/stream.go:1267]: The function `getGlobalResolvedTS` seems to be no longer used.

**Review by 3pointer — APPROVED** [review]
LGTM

**Pull request merged**